src/darwin/Framework/CHIP/MTRDeviceController_Concrete.mm

```diff
@@ -449,6 +449,13 @@ - (void)cleanupAfterStartup
     for (MTRDevice * device in devices) {
         [device invalidate];
     }
+
+    // Since MTRDevice invalidate may issue asynchronous writes to storage, perform a
+    // block synchronously on the storage delegate queue to flush those operations.
```

Comment:
This comment still doesn't explain why we need to block _here_ on that.  To repeat my review comment, which was marked resolved without resolving:

> What happens if we don't do that? Won't the writes happen anyway? Or are we worried about our caller tearing down the storage backend if we return from here, before the writes have a chance to happen? Would be good to document why this needs to be a sync point.

Reply:
Ah I didn't see it before the previous comment was resolved. And yes you were right in that I was worried about storage backend going away after shutdown returns. Let me clarify a bit more.